### PR TITLE
Research general improvements

### PR DESCRIPTION
**Pull request by Brett208**
This branch should resolve lingering pre-merge issues from branch #21 

@TechCor8, would you mind spot testing some of the reworked functions to make sure I didn't damage anything. I compiled and ran a map using HFL, but I didn't specifically test and of the research functions. Actually, it isn't clear to me how Research.h would be accessed/used by a mission.

I was a bit liberal editing the comments in TechCategory. Please let me know if you want them reverted.

### CODE REVIEW

File: Source/Research.cpp [hunk 1]
```diff
@@ -39,159 +38,179 @@ OP2Research *researchObj;
...
-	OP2TechInfo *p = (OP2TechInfo*)internalPtr;
-	return p->numRequiredTechs;
+	return internalPtr->numRequiredTechs;
 }
+
 char* TechInfo::GetTechName()
 {
-	if (!isInited)
+	if (!isInited) {
 		return (char*)HFLNOTINITED;
```
Comment (Brett208) [inline Source/Research.cpp:176]: I wasn't sure which cast was appropriate for this return so I just left it as a c-style cast for now...

File: Source/Research.h [hunk 2]
```diff
@@ -5,27 +5,27 @@
 
 enum LabType
 {
-	ltBasic			= 1,
-	ltStandard		= 2,
-	ltAdvanced		= 3
+	ltBasic	= 1,
+	ltStandard = 2,
+	ltAdvanced = 3
 };
```
Comment (DanRStevens) [inline Source/Research.h:9]: This is kind of a separate issue, though I wanted to mention it here. We should probably use `enum class` in place of `enum`. As `enum class` has mandatory scoping (`LabType::ltBasic`), we should also remove the `lt` prefix on each enum value name.

File: Source/Research.cpp (hunk 1 above)
Comment (DanRStevens) [reply Source/Research.cpp:176]: It's fine. Issue #31 is set to remove all such casts anyway. Let's just deal with it then.

**Review by DanRStevens — APPROVED** [review]
Looks good.

**Pull request merged**